fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -938,6 +938,58 @@ private void analyzeAddPartition(OlapTable olapTable, List<PartitionDesc> partit
         }
     }
 
+    private int calBucketNumAccordingToBackends() {
+        int backendNum = GlobalStateMgr.getCurrentSystemInfo().getBackendIds().size();
+        // When POC, the backends is not greater than three most of the time.
+        // The bucketNum will be given a small multiplier factor for small backends.
+        int bucketNum = 0;
+        if (backendNum <= 3) {
+            bucketNum = 2 * backendNum;
+        } else if (backendNum <= 6) {
+            bucketNum = (int) (1.5 * backendNum);
+        } else if (backendNum <= 12) {
+            bucketNum = (int) (1.2 * backendNum);
+        } else {
+            bucketNum = backendNum;
+        }
+        return bucketNum;
+    }
+
+    private int calAvgBucketNumOfRecentPartitions(OlapTable olapTable, int recentPartitionNum) {
+        // 1. If the partition is less than recentPartitionNum, use backendNum to speculate the bucketNum
+        int bucketNum = 0;
```

Comment:
Why use the default, instead of all existing Partitions, to estimate when it is less than 5?